galasa-ui/src/actions/getTestRuns.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+'use server';
+
+import { ResultArchiveStoreAPIApi, Run, RunResults } from "@/generated/galasaapi";
+import { createAuthenticatedApiConfiguration } from "@/utils/api";
+import * as Constants from "@/utils/constants";
+
+/**
+ * Fetches test runs from the Result Archive Store (RAS) for the last 24 hours.
+ * 
+ * @param {string} loginId - The login ID of the user.
+ * @returns {Promise<Run[]>} - A promise that resolves to an array of Run objects.
+ */
+export const fetchMyTestRunsForLastDay  = async (): Promise<Run[]> => {
```

Comment:
This function is defined as a server action, which is typically more suited towards updating/mutating data (it sends a POST request to the server behind the scenes). See https://nextjs.org/docs/app/getting-started/updating-data.

Maybe we can define this function in the TestRunsPage, pass down the Run[] data into the client component, and use a suspense or [loading.tsx](https://nextjs.org/docs/app/api-reference/file-conventions/loading) to render an initial skeleton while the server fetches the data?